### DISCUSSION TITLE
Generalized OMR/OCR Layout

### DISCUSSION THREAD

**Discussion by karsood** [Sunbird-Saral/Community#22] [Ideas]
Context: In conversations with multiple organizations and state departments, we have had to get into a few rounds of discussions to finalize the layout that will be used for student assessments. It takes a few rounds of discussions because either the state department has a design of a question paper which can not be catered (100+ questions per student) or that they want to do it for multiple students but the question paper blueprint becomes hard to incorporate given the layout constraints.

Desire: To reduce these conversations and to give a heads up to the state departments before they go about creating the question paper blueprint, is it possible to tell them upfront the most number of questions we can incorporate in our layout? Example - the layout will be able to handle 19 columns x 10 rows for an OCR (student ID) + OMR (question response) layout to be filled by the teacher.

Request: A strict framework (not a suggestive guideline) to clearly communicate to the state department beforehand the following:
1. Rows and Columns that can be accommodated for purely OCR sheet (marks entered will require 1 box) and hence the number of questions that they can include in their question paper against X digits in the student ID.
2. Rows and Columns that can be accommodated for purely OCR sheet (marks entered will require 2 boxes) and hence the number of questions that they can include in their question paper against X digits in the student ID.
3. Rows and Columns that can be accommodated for OMR (question response) + OCR (student ID) sheet  and hence the number of questions that they can include in their question paper against X digits in the student ID.
4. Rows and Columns that can be accommodated for student-filled MCQ sheet and hence the number of questions that they can include in their question paper against X digits in the student ID.

Other considerations:
1. If a state has a requirement for more number of questions then an accommodation for multi-row should be made.
2. If a state has a requirement to adjust student IDs, then that could be done by reducing space given to questions

Purpose: If we solidify the layout design then taking it to states will become easier and we will reduce the time taken to finalize the layout design as per the blueprint. In addition, we can use the generalize layout to make modifications and develop the most optimal size (rows vs columns) in future versions.

**Comment by dileep-gadiraju** [comment]
Agree with you @karsood , will work with the designer and we will come up with an outline first. Post that we can discuss and come up with generalized layouts as discussed.
